runtime/src/bank.rs

```diff
@@ -4705,8 +4693,27 @@ impl Bank {
                         timings,
                         &mut error_counters,
                         log_messages_bytes_limit,
-                        tx_executor_cache.clone(),
-                    )
+                        programs_loaded_for_tx_batch.clone(),
+                    );
+
+                    if let TransactionExecutionResult::Executed {
```

Comment:
This could be factored into a `merge_caches()` method. Both here where the TX goes into the batch, and also later where the batch goes into the global cache.